commerce_coordinator/apps/lms/views.py

```diff
@@ -401,3 +412,121 @@ def post(self, request):
             logger.exception(f"[FirstTimeDiscountEligibleView] HTTP Error: {err}")
 
         return Response({'is_eligible': True})
+
+
+class ProgramPriceView(APIView):
+    """View to get the price of a program."""
+    authentication_classes = (JwtAuthentication,)
+    permission_classes = (AllowAny,)
+    throttle_classes = []
+
+    def _calculate_discounted_price(self, program_offer, program_price):
+        """Calculate the discounted price of the program."""
+        if not program_offer:
+            return program_price
+
+        discounted_price = program_price
+        if program_offer.get("discount_type") == CT_ABSOLUTE_DISCOUNT_TYPE:
+            discounted_price -= program_offer.get("discount_value_in_cents")
+        else:
+            discounted_price -= discounted_price * program_offer.get("discount_value_in_cents") / 10000
+
+        return discounted_price
+
+    def _extract_course_id(self, course_string):
+        """
+        Extract course_id for comparision
+        """
+        match = re.search(r"course-v1:(.+)", course_string)
+        return match.group(1) if match else None
+
+    def _filter_enrollable_program_entitlements(self, program_variants, course_keys, enrollments, entitlements):
+        """Filter out bundle entitlements in which user can enroll."""
+
+        active_user_enrollments = []
+        for usr_enr in enrollments:
+            if usr_enr.get("is_active") and usr_enr.get("mode") not in EXCLUDED_ENROLLMENT_COURSE_MODES:
+                course_key = usr_enr.get("course_details", {}).get("course_id")
+                active_user_enrollments.append(self._extract_course_id(course_key))
+
+        active_user_entitlements = []
+        for usr_ent in entitlements:
+            if not usr_ent.get("expired_at"):
+                course_key = usr_ent.get("course_details", {}).get("course_id")
+                active_user_entitlements.append(usr_ent.get("course_uuid"))
+
+        filtered_entitlement_skus = []
+        filter_out_course_keys = course_keys + active_user_enrollments + active_user_entitlements
+        for variant in program_variants:
+            course_key = variant.get("course_key")
+            if course_key not in filter_out_course_keys:
+                filtered_entitlement_skus.append(variant.get("standalone_price_sku"))
```

Comment:
I am confused on this part, can we connect on this?

Reply:
sure, we can connect.